app/views/admin/navigation_links/_form.html.erb

```diff
@@ -8,10 +8,21 @@
   <%= form.text_field :url, class: "form-control" %>
   <div class="alert alert-info">A full (external) or relative (internal) URL for the link</div>
 </div>
-<div class="form-group">
-  <%= form.label :icon %>
-  <%= form.text_area :icon, class: "form-control" %>
-  <div class="alert alert-info">An SVG icon should be pasted in the textarea</div>
+<div class="form-group" data-controller="svg-icon-upload">
+  <div data-svg-icon-upload-target="navId" nav-link-id="<%= form.object[:id] %>">
+    <%= form.label :icon %>
+    <div data-svg-icon-upload-target="svgIconPreview" class="<%= form.object[:id].nil? ? "" : "pb-3" %>" style="width: 48px;">
+      <%= form.object[:icon]&.html_safe %>
+    </div>
+    <%= form.hidden_field :icon, "data-svg-icon-upload-target": "svgIconContent" %>
+
+    <input
+      type="file" accept="image/svg+xml" class="form-control"
```

Comment:
```suggestion
      type="file" accept="image/svg+xml" class="form-control-file"
```

Refactor/blocking - I think the normal input styling looks a little funky with the file upload option - on Chrome and Safari in particular we have spacing above the button but not below, and the border maybe isn't suitable for this input type. 

I suggest we change to the styling for `form-control-file` which gives us something more specific to this input type.

Before on Chrome:

![screenshot showing button with spacing at top but not bottom](https://user-images.githubusercontent.com/20773163/106888026-e0c8e580-66dd-11eb-89b2-8c03ec10e6d8.png)

After:

![screenshot showing a more styled button without the bordered input](https://user-images.githubusercontent.com/20773163/106888074-f1795b80-66dd-11eb-97bd-523def28856e.png)

Reply:
Done @aitchiss!